test/udp_sender.js

```diff
@@ -249,4 +251,20 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', (done) => {
+        let spy = sinon.spy(console, 'log');
+        sender._host = 'foo.bar.com';
+        new Tracer(
+            'test-service-name',
+            new RemoteReporter(sender),
+            new ConstSampler(true)
+        ).startSpan('testSpan').finish();
+        sender.flush();
+        setTimeout(() => {
```

Comment:
*NOTE*: `setTimeout` is needed due to Node's asynchronous nature. Because I am spying on `console.log`, if `setTimeout` is not used, the error message we expect will be printed to the console but not before mocha prints other messages to the console.